packages/discovery-react-components/src/components/DocumentPreview/__tests__/DocumentPreview.test.tsx

```diff
@@ -28,91 +74,265 @@ describe('DocumentPreview', () => {
-      extracted_metadata: {
-        filename: 'i_am_a_file'
-      },
-      text: ['Example text <text> <username> <password> more text afterwards']
-    };
+  // All of these currently fail since DocumentPreview defaults to HtmlView if a HTML field is present.
+  // Also fails because SimpleDocument will not show a JSON document without a passage highlight.
+  describe.skip('JSON files', () => {
+    it('renders text field for JSON files', async () => {
+      act(() => {
+        ({ getByText } = render(<DocumentPreview document={jsonDoc} />));
+      });
+      getByText('Hello, I am a text field.');
+    });
 
-    let getByText: NonNullable<any>;
+    it('renders an overwritten "body" field for JSON files', async () => {
+      context = {
+        componentSettings: {
+          fields_shown: {
+            body: {
+              field: 'body_field'
+            }
+          }
+        }
+      };
+      act(() => {
+        ({ getByText } = render(
+          wrapWithContext(<DocumentPreview document={jsonDoc} />, api, context)
+        ));
+      });
+      getByText('Display text from the specified "body" field.');
+    });
 
-    act(() => {
-      ({ getByText } = render(<DocumentPreview document={htmlTextDoc} />));
+    it('renders text from passage field, with passage highlighted', async () => {
+      const highlight = {
+        passage_text: 'Hello',
+        start_offset: 0,
+        end_offset: 4,
+        field: 'text'
+      };
+
+      act(() => {
+        ({ findAllByTestId } = render(
+          <DocumentPreview document={jsonDoc} highlight={highlight} />
+        ));
+      });
+
+      const highlights = await findAllByTestId('field-rect');
+      expect(highlights.length).toEqual(1);
+      const highlightContainer = highlights[0].parentElement;
+      expect(highlightContainer.getAttribute('data-field-type')).toEqual('passage');
+      expect(highlightContainer.getAttribute('data-field-id')).toEqual('0');
     });
+  });
 
-    getByText('<text> <username> <password>', { exact: false });
+  describe('CSV files', () => {
+    // Currently fails since no error message is displayed if there is an unspecified body field and passage
+    it.skip('shows error if no "body" field or passage have been specified', () => {
+      act(() => {
+        ({ getByText } = render(<DocumentPreview document={csvDoc} />));
+      });
+      getByText('Cannot preview document');
+    });
+
+    it('renders specfied "body" field', () => {
+      context = {
+        componentSettings: {
+          fields_shown: {
+            body: {
+              field: 'column_1'
+            }
+          }
+        }
+      };
+      act(() => {
+        ({ getByText } = render(
+          wrapWithContext(<DocumentPreview document={csvDoc} />, api, context)
+        ));
+      });
+      const simpleDoc = getByText('This is column 1.').parentElement;
+      expect(simpleDoc.classList.contains('bx--simple-document__content')).toBe(true);
```

Comment:
This `expect` is also an issue (see comment above on similar assertion for JSON files).